modules/vpc/user_management/main.tf

```diff
@@ -181,3 +181,62 @@ resource "aws_lambda_permission" "allow_cloudwatch" {
   principal     = "events.amazonaws.com"
   source_arn    = "${aws_cloudwatch_event_rule.lambda.arn}"
 }
+
+# TODO: Test if we can run config file with less config options
+# that way we can don't have to generate the full config
+resource template_file "user_management_config_iam" {
+    count   = "${var.enabled}"
+
+    # We are just using the template in the previous directory for now
+    template = "${file("${path.module}/../user_management.yml.tmpl")}"
+
+    lifecycle {
+        create_before_destroy = true
+    }
+
+    vars {
+        region              = "${var.region}"
+        environment         = "global"
+        smtp_from_address   = "${var.user_management_smtp_from_address}"
+        smtp_username       = "${var.user_management_smtp_username}"
+        smtp_password       = "${var.user_management_smtp_password}"
+        smtp_host           = "${var.user_management_smtp_host}"
+        smtp_port           = "${var.user_management_smtp_port}"
+        ldap_server         = "${var.user_management_ldap_server}"
+        ldap_port           = "${var.user_management_ldap_port}"
+        ldap_base_dn        = "${var.user_management_ldap_base_dn}"
+        ldap_bind_user      = "${var.user_management_ldap_bind_user}"
+        ldap_bind_password  = "${var.user_management_ldap_bind_password}"
+        tls_cert            = "${replace(file("${path.cwd}/${var.user_management_tls_cert}"), "/\\n/", "\n    ")}"
+        tls_key             = "${replace(file("${path.cwd}/${var.user_management_tls_cert}"), "/\\n/", "\n    ")}"
+        global_admin_ldap_group     = "${var.user_management_global_admins}"
+        sudo_user_ldap_group        = "${var.user_management_sudo_users}"
+        users_ldap_group            = "${var.user_management_users}"
+    }
+}
+
+resource "null_resource" "user_management_unicreds_iam" {
+    count = "${var.enabled}"
+
+    lifecycle {
+        create_before_destroy = true
+    }
+
+    triggers {
+        region              = "${var.region}"
+        context             = "-E region:${var.region} -E environment:global -E service:nubis"
+        rendered_template   = "${template_file.user_management_config_iam.rendered}"
+        credstash           = "unicreds -r ${var.region} put-file nubis/global"
+    }
+
+    # Ugly, I can't just do a put with unicreds, and I have to use unicreds because
+    # credstash seems to complain when I use it wit this error:
+    # An error occurred (ValidationException) when calling the Query operation: One or more parameter values were invalid: Size of hashkey has exceeded the maximum size limit of2048 bytes
+    provisioner "local-exec" {
```

Comment:
Avoid temporary local files as much as possible. In this case, found out that unicreds doesn't dig '-' as a filename, but a workaround like this one *will* work:

echo "${template_file.user_management_config_iam.rendered}" | ${self.triggers.credstash}/user-sync/config /dev/stdin ${self.triggers.context}